src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -292,10 +335,19 @@ DatabaseShardImp::prepareShard(std::uint32_t shardIndex)
         return false;
     }
 
-    // Check available storage space
-    if (fileSz_ + avgShardFileSz_ > maxFileSz_)
-        return fail("maximum storage size reached");
-    if (avgShardFileSz_ > available())
+    // Any shard earlier than the two most recent shards
+    // is a historical shard
+    bool const isHistoricalShard = shardIndex < shardBoundaryIndex(lock);
+    auto const numHistoricalShards = this->numHistoricalShards(lock);
```

Comment:
Now that we've renamed the function, we should probably rename this variables. Maybe `numHistShards`.

Reply:
Done